engine/runtime/src/test/scala/org/enso/compiler/test/pass/analyse/AliasAnalysisTest.scala

```diff
@@ -833,6 +833,63 @@ class AliasAnalysisTest extends CompilerTest {
     }
   }
 
+  "Alias analysis on self" should {
+    implicit val ctx: ModuleContext = mkModuleContext
+
+    val addMethod =
+      """type Foo
+        |    type Foo a b
+        |    add x = self.a + x
+        |""".stripMargin.preprocessModule.analyse
+        .bindings(2)
+        .asInstanceOf[Method.Explicit]
+
+    val graph = addMethod
+      .unsafeGetMetadata(AliasAnalysis, "Missing aliasing info")
+      .unsafeAs[Info.Scope.Root]
+      .graph
+    val graphLinks = graph.links
+
+    val lambda = addMethod.body.asInstanceOf[IR.Function.Lambda]
+
+    "assign Info.Scope.Root metadata to the method" in {
+      val meta = addMethod.getMetadata(AliasAnalysis)
+      meta shouldBe defined
+    }
+
+    "not add self to the scope" in {
+      lambda.arguments.length shouldEqual 2
+      lambda.arguments(0).name shouldBe a[IR.Name.Self]
```

Comment:
What is this test testing? Is this the _syntetic self_? As I don't see any `self` in the `add` method definition.

Reply:
Yes, this makes sure that _synthetic_ self isn't added to the scope, which is done in AliasAnalysis